src/Engine/GameInput/GameInputState.java

```diff
@@ -21,8 +21,8 @@
   /** The current GameActionBuilder state data,
    *  shared with all State instances. */
   protected final StateData myStateData;
-  protected final OptionSet myOptions;
-  protected final OptionSelector mySelector;
+  protected OptionSet myOptions;
```

Comment:
These kinds of changes make me nervous. Also, I re-added final and it looks like it still works.

Reply:
That was done under `a quick hack to keep things rolling`.
I was using the same inputstate for both quit and exit, and the only way I could figure out to make that work (without too much effort) was to make it no longer final so I could re-set it in the constructor of the sub-class.
I believe it's no longer needed since I ended up making them into two separate classes.

Full commit message:
```
Quick hack to keep things rolling
I feel like we need a better design for this
I really just wish we could set subclass fields up before super()
```

Reply:
> I was using the same inputstate for both quit and exit

Haven't those always been separate? `ConfirmExit` either advances to `ExitToMainMenu` or straightforwardly terminates the application.

> I really just wish we could set subclass fields up before super()

Yeah, that _is_ one downside of the current approach. The thought is that `myStateData` should have all information necessary to initialize `myOptions`, but in practice there are many kinds of state info, so it can end up being annoying. I've had a thought or two on how to address that, but so far I haven't had a use case that has forced me to change things.